pyproject.toml

```diff
@@ -53,6 +53,7 @@ requests = "^2.32.3"
 aiohttp = { version = "^3.9.1", optional = true}
 aiohttp-sse-client = { version = "^0.2.1", optional = true}
 mnemonic = "^0.20"
+shamir-mnemonic = "^0.3.0"
```

Comment:
I suggest setting it as an optional dependency, as SLIP 39 does not seem to be widely used yet.
```
shamir-mnemonic = { version = "^0.3.0", optional = true }

[tool.poetry.extras]
aiohttp = ["aiohttp", "aiohttp-sse-client"]
shamir = ["shamir-mnemonic"]
```

Reply:
I added the import in each function with a check. This way there is no global import. I can also use a try except block and a `HAS_SHAMIR` variable if you prefer that.

Reply:
👍 Your approach is great and doesn't need any changes.